ui/lib/pki/addon/components/pki-role-generate.ts

```diff
@@ -0,0 +1,87 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { service } from '@ember/service';
+import Router from '@ember/routing/router';
+import Store from '@ember-data/store';
+import { tracked } from '@glimmer/tracking';
+import errorMessage from 'vault/utils/error-message';
+import FlashMessageService from 'vault/services/flash-messages';
+import DownloadService from 'vault/services/download';
+
+interface Args {
+  onSuccess: CallableFunction;
+  model: PkiCertificateGenerateModel;
+}
+
+interface PkiCertificateGenerateModel {
```

Comment:
TS question. If the model was typescript would this interface belong in that file instead?

Reply:
Yes, ideally once the models are TS we can just import from there! For now we're just making the component happy :)